src/black/linegen.py

```diff
@@ -815,6 +815,19 @@ def _first_right_hand_split(
     tail_leaves.reverse()
     body_leaves.reverse()
     head_leaves.reverse()
+
+    if Preview.hug_parens_with_braces_and_square_brackets in line.mode:
+        if (
+            tail_leaves[0].type == token.RPAR
+            and tail_leaves[0].value
+            and tail_leaves[0].opening_bracket is head_leaves[-1]
```

Comment:
```suggestion
            and tail_leaves[0].opening_bracket == head_leaves[-1]
```
I don't think we can guarantee identity

Reply:
Why not? The whole `omit` system in `_first_right_hand_split` seems to rely on `id(leaf)`, so the leaf objects should be stable and thus identity comparisons should be safe also here. Or am I missing something?

Reply:
Thanks, I was confused